src/main/java/com/redhat/foreman/ForemanSharedNodeCloud.java

```diff
@@ -526,7 +528,8 @@ void updateHostData() {
         hostsMap.set(new HashMap<String, String>());
     }
 
-    private Map<String, String> getHostsMapData() {
+    @Restricted(DoNotUse.class) // index.jelly
+    public Map<String, String> getHostsMapData() {
```

Comment:
I guess we need another method for exposing compatible hosts publicly. 

This method can return false data - for ex. until the cloud is in operational state (aka after startup cleanup), this return **empty** map, the same if network issue occurred, here you can only see empty map and can't get an idea what went wrong.

So I'd prefer to leave this method private and for internal purpose only and implement the new one which contacts the cloud provider (Foreman instance) directly (via ForemanApi.getCompatibleHosts()) and can report correct data and issues to Jelly template too.